modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/evaluator/RuntimeEvaluator.scala

```diff
@@ -0,0 +1,507 @@
+    }
+  }
+}
+
+/* -------------------------------------------------------------------------- */
+
+private object MatchingMethods {
+  def fromLitToValue(frame: JdiFrame, literal: Lit): JdiValue = {
+    val vm = frame.current().virtualMachine()
+    val value = literal match {
+      case Lit.Boolean(value) => vm.mirrorOf(value)
+      case Lit.Byte(value) => vm.mirrorOf(value)
+      case Lit.Char(value) => vm.mirrorOf(value)
+      case Lit.Double(value) => vm.mirrorOf(value.toDouble)
+      case Lit.Float(value) => vm.mirrorOf(value.toFloat)
+      case Lit.Int(value) => vm.mirrorOf(value)
+      case Lit.Long(value) => vm.mirrorOf(value)
+      case Lit.Short(value) => vm.mirrorOf(value)
+      case Lit.String(value) => vm.mirrorOf(value)
+      case Lit.Unit() => vm.mirrorOfVoid()
+      case _ => throw new IllegalArgumentException(s"Unsupported literal: $literal")
+    }
+
+    JdiValue(value, frame.thread)
+  }
+
+  /* -------------------------------------------------------------------------- */
+  /*                                Method lookup                               */
+  /* -------------------------------------------------------------------------- */
+  private def argsMatch(method: Method, args: Seq[Type], frame: JdiFrame): Boolean =
+    method.argumentTypeNames().size() == args.size && sameTypes(method, args, frame)
+
+  /**
+   * Look for a method with the given name and arguments types, on the given reference type
+   *
+   * Encode the method name by default, but can be disabled for methods that are not Scala methods (such as Java <init> methods)
+   *
+   * If multiple methods are found, a [[ValidationMonad.Unrecoverable]] is returned
+   *
+   * Also, if the method return type is not loaded or prepared, it will be loaded and prepared
+   *
+   * @param ref the reference type on which to look for the method
+   * @param funName the name of the method
+   * @param args the arguments types of the method
+   * @param frame the current frame
+   * @param encode whether to encode the method name or not
+   * @return the method, wrapped in a [[ValidationMonad]], with its return type loaded and prepared
+   */
+  def methodsByNameAndArgs(
+      ref: ReferenceType,
+      funName: String,
+      args: Seq[Type],
+      frame: JdiFrame,
+      encode: Boolean = true
+  ): Validation[Method] =
+    ref
+      .methodsByName { if (encode) NameTransformer.encode(funName) else funName }
+      .asScalaSeq
+      .filter(method => !method.isPrivate && argsMatch(method, args, frame))
+      .toSeq
+      .toValidation(s"Cannot find methods $funName with args types $args")
+      .map(loadClassOnNeed(_, frame))
+
+  /* -------------------------------------------------------------------------- */
+  /*                                Type checker                                */
+  /* -------------------------------------------------------------------------- */
+  def sameType(got: Type, expected: Type, frame: JdiFrame): Boolean = {
```

Comment:
`isAssignableFrom` is probably a better name.

```suggestion
  def isAssignableFrom(got: Type, expected: Type, frame: JdiFrame): Boolean = {
```